app/classes/Typolib/Typolib/Rule.php

```diff
@@ -39,4 +63,38 @@ public static function displayIfThenArrayRule()
             echo "Input character: $key => New character: $value<br />\n";
         }
     }
+
+    public static function generateRuleId()
+    {
+        $array = Rule::scanDirectory(DATA_ROOT . 'code');
+        if (empty($array)) {
+            $id = 0;
+        } else {
+            $id = max($array);
+        }
+
+        return ++$id;
+    }
+
+    public static function scanDirectory($dir)
+    {
+        global $allIds;
+
+        if (is_dir($dir)) {
+            $me = opendir($dir);
+            while ($child = readdir($me)) {
+                if ($child != '.' && $child != '..') {
+                    if ($child == 'rules.php') {
+                        include $dir . DIRECTORY_SEPARATOR . $child;
```

Comment:
Uh, I guess we should be using DIRECTORY_SEPARATOR constant everywhere to become OS-agnostic…